### PR TITLE
Minor grammatical corrections and improvements

### PR DESCRIPTION
**Pull request by PabloL007**
This proposal includes minor grammatical corrections to `02-Design/01-Principles of Security Engineering.md`, as well as, the addition of an excerpt from "Basic Principles of Information Protection" that I found useful for better understanding the least common mechanism principle.

### CODE REVIEW

**Comment by PabloL007** [comment]
@kapilepatel I see you reviewed the last PR that was approved could you take a look a this one?

**Comment by PabloL007** [comment]
Maybe @bradchesney79 ?

File: 02-Design/01-Principles of Security Engineering.md [hunk 1]
```diff
@@ -18,22 +18,22 @@ Limits the damage in case of exploited vulnerability.
 In order to apply this principle, proper granularity of privileges and permissions should be established.
 
 ## Separation of Duties
-Also known as the [compartmentalization principle][1], or separation of privilege, separation of duties is a security principle which states that the successful completion of a single task is dependent upon two or more conditions that need to be met and just one of the conditions will be insufficient in completing the task by itself.
+Also known as the [compartmentalization principle][1], or separation of privilege, separation of duties is a security principle which states that the successful completion of a single task is dependent upon two or more conditions that are insufficient for completing the task by themselves.
```
Comment (bradchesney79) [inline 02-Design/01-Principles of Security Engineering.md:21]: I like most of it, but completion of a single task doesn't agree with the plural of themselves.

**Review by bradchesney79 — CHANGES_REQUESTED** [review]
Line 21, themselves to itself. Plural direct object doesn't agree with singular subject. On the whole, nice edits otherwise. Thank you for your contribution.

File: 02-Design/01-Principles of Security Engineering.md (hunk 1 above)
Comment (PabloL007) [reply 02-Design/01-Principles of Security Engineering.md]: True, changed back to itself and pushed

**Comment by PabloL007** [comment]
> Line 21, themselves to itself. Plural direct object doesn't agree with singular subject. On the whole, nice edits otherwise. Thank you for your contribution.

Happy to help, I changed it back to itself, let me know if there’s anything else

**Review by bradchesney79 — APPROVED** [review]

**Pull request merged**